### PR TITLE
 fix(ui): admins see join button for public teams

### PR DESCRIPTION
**Pull request by rakdutta**
closes issue #3488 
This pull request fixes a bug where platform administrators viewing public teams they are not members of would see admin controls (like "Manage Members" and "Delete Team") instead of the "Request to Join" button. The change ensures that admin users must follow the same join request workflow as regular users for public teams, while still preserving emergency admin controls for private teams. The fix involves reordering the conditional logic that determines which UI elements are shown, and includes new unit test coverage for this scenario.

**Bug Fix: Admin Join Button for Public Teams**

*Logic Correction and UI Consistency:*
- Reordered the conditional logic in `admin_list_teams_partial()` within `mcpgateway/admin.py` so that public team status is checked before admin status. This ensures that admins who are not team members see the "Request to Join" button for public teams, rather than admin controls. Admin controls remain available for private teams to allow for emergency access and maintenance.

*Testing:*
- Introduced a new unit test, `test_admin_sees_join_button_for_public_teams`, in `tests/unit/mcpgateway/test_admin.py` to verify that platform admins see the correct "public" relationship (and thus the "Request to Join" button) for public teams, and "none" (admin controls) for private teams. This ensures the bug is fixed and prevents regressions.
- Added integration tests in `tests/integration/test_admin_teams_ui.py` to verify the full stack behavior with actual HTML rendering and database queries.

### CODE REVIEW

File: tests/unit/mcpgateway/test_admin.py [hunk 1]
```diff
@@ -23372,3 +23372,114 @@ async def test_admin_list_teams_forwards_search_query(self, monkeypatch, mock_ad
             call_kwargs = mock_team_service.list_teams.call_args[1]
             assert call_kwargs["search_query"] == "nomatch"
             assert call_kwargs["personal_owner_email"] == "admin@example.com"
```
Comment (prakhar-singh1928) [inline tests/unit/mcpgateway/test_admin.py:23558]: @rakdutta Thank you for your submission. A couple of test cases are missing to ensure complete coverage:

**Missing Test 1:** test_admin_member_sees_member_controls_not_join_button

**Scenario:** An admin user who is already a member of a public team
**Expected Result:** They should see normal member controls (such as "Leave Team"), NOT the "Request to Join" button

**Missing Test 2:** test_admin_sees_pending_request_status_for_public_teams

**Scenario:** An admin user who has requested to join a public team but hasn't been approved yet
**Expected Result:** The UI should display "Request Pending" status, not the "Request to Join" button
These tests will verify that the fix handles all edge cases correctly and prevent regressions.

Comment (rakdutta) [reply tests/unit/mcpgateway/test_admin.py:23558]: @prakhar-singh1928 Thanks for your review comment.

 Added both test cases as requested:                                                                                                                                                                                                               
                                                                                                                                                                                                                                                    
  1. **test_admin_member_sees_member_controls_not_join_button**: Verifies that an admin who is already a member of a public team sees member controls (relationship='member') rather than the join button. This confirms the membership check takes 
  precedence over the public visibility check.                                                                                                                                                                                                      
                                                                                                                                                                                                                                                    
  2. **test_admin_sees_pending_request_status_for_public_teams**: Verifies that an admin with a pending join request for a public team sees 'Request Pending' status (relationship='public' with pending_request populated) rather than the join    
  button.                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                    
  Both tests pass and confirm the existing logic correctly handles these edge cases. The implementation in mcpgateway/admin.py:5448-5464 already handles these scenarios properly:                                                                  
  - Line 5452: Membership is checked first                                                                                                                                                                                                          
  - Line 5455-5460: Public team logic (including pending request tracking)                                                                                                                                                                          
  - Line 5461-5464: Admin controls only for non-public teams                                                                                                                                                                                        
                                                                                                                                                                                                                                                    
  All 11 tests in TestAdminPersonalTeamFiltering pass.

**Comment by prakhar-singh1928** [comment]
All the issues are now addressed.
the PR looks good to me.

**Review by prakhar-singh1928 — DISMISSED** [review]
